hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -650,15 +653,26 @@ public void deregisterAllListener(String cacheNameWithPrefix) {
     @Override
     public Map<String, LocalCacheStats> getStats() {
         Map<String, LocalCacheStats> stats = createHashMap(statistics.size());
+
         for (Map.Entry<String, CacheStatisticsImpl> entry : statistics.entrySet()) {
-            stats.put(entry.getKey(), new LocalCacheStatsImpl(entry.getValue()));
+            String cacheName = entry.getKey();
+            LocalCacheStatsImpl localCacheStats = new LocalCacheStatsImpl(entry.getValue());
+            CompletableFuture<CacheConfig> configFuture = configs.get(cacheName);
+            if (configFuture.isDone()) {
+                try {
+                    localCacheStats.setNativeMemoryUsed(configFuture.get().getInMemoryFormat() == NATIVE);
+                } catch (InterruptedException | ExecutionException e) {
+                    e.printStackTrace();
+                }
+            }
+            stats.put(cacheName, localCacheStats);
         }
         return stats;
     }
 
     @Override
     public CacheOperationProvider getCacheOperationProvider(String cacheNameWithPrefix, InMemoryFormat inMemoryFormat) {
-        if (InMemoryFormat.NATIVE.equals(inMemoryFormat)) {
+        if (NATIVE.equals(inMemoryFormat) && !BuildInfoProvider.getBuildInfo().isEnterprise()) {
```

Comment:
What does this change have to do with sending stats to MC?

Reply:
Without this line the `TimedMemberStateFactoryTest#nativeMemoryEnabledForCache()` test fails with an exception with this stacktrace:
```
java.lang.IllegalArgumentException: Native memory is available only in Hazelcast Enterprise.Make sure you have Hazelcast Enterprise JARs on your classpath!

	at com.hazelcast.cache.impl.AbstractCacheService.getCacheOperationProvider(AbstractCacheService.java:677)
	at com.hazelcast.cache.impl.CacheProxySupport.<init>(CacheProxySupport.java:121)
	at com.hazelcast.cache.impl.CacheProxy.<init>(CacheProxy.java:95)
	at com.hazelcast.cache.impl.AbstractCacheService.createDistributedObject(AbstractCacheService.java:270)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.doCreateProxy(ProxyRegistry.java:219)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.createProxy(ProxyRegistry.java:211)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.getOrCreateProxyFuture(ProxyRegistry.java:178)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyRegistry.getOrCreateProxy(ProxyRegistry.java:159)
	at com.hazelcast.spi.impl.proxyservice.impl.ProxyServiceImpl.getDistributedObject(ProxyServiceImpl.java:148)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.getDistributedObject(HazelcastInstanceImpl.java:352)
	at com.hazelcast.cache.impl.HazelcastInstanceCacheManager.getCacheByFullName(HazelcastInstanceCacheManager.java:51)
	at com.hazelcast.cache.impl.HazelcastInstanceCacheManager.getCache(HazelcastInstanceCacheManager.java:45)
	at com.hazelcast.internal.management.TimedMemberStateFactoryTest.nativeMemoryEnabledForCache(TimedMemberStateFactoryTest.java:94)
```

So I was looking for a workaround to overcome this hard check. Previously I saw that the `ConfigValidator#checkNotNativeWhenOpenSource()` method (which runs for every map and cache config) uses this `BuildInfo#isEnterprise()` flag for validation, and this flag is controllable using the `HAZELCAST_INTERNAL_OVERRIDE_ENTERPRISE` system property (that's why we toggle it in the `@Before` and the `@After` methods). So I needed to work this hard check around and I thought it makes sense to stick to the approach that is already utilized in an other part of the codebase (config validation).

Reply:
Thanks for the detailed explanation !